astsearch.py

```diff
@@ -462,6 +462,8 @@ def main(argv=None):
                     help="AST pattern to search for; see docs for examples")
     ap.add_argument('path', nargs='?', default='.',
                     help="file or directory to search in")
+    ap.add_argument('-l', '--files-with-matches', action='store_true',
+                    help="only print the paths with at least one match")
```

Comment:
I see from the code what this does, but I'd like to find a better wording for the help string. This sounds to me like the default is to print all paths, and this option skips printing paths that had no matches. What about something like:

> Output only the paths of matching files, not the lines that matched

I'd also prefer to drop the short option `-l`, because it doesn't seem clearly connected. I guess it's there for parity with grep? I don't really want to copy the grep interface - especially if it leads to people trying to add all the many options from grep to this code.

Reply:
Rebased and edited the help message.  The -l option is indeed present in grep, as well as the numerous variants git-grep, ag, rg, etc, and --files-with-matches is rather a mouthful, so I'd rather keep it.
But if you confirm you don't want it, I'll kill it.

Reply:
OK, on further thought, it does sound like a relatively common thing to do, so I'll let the short form pass.